### ISSUE TITLE
Add support for Mercurial metadata

### ISSUE THREAD

**Issue by sjl** [topfunky/PeepOpen-Issues#91]
This would be really handy for those of use that use Mercurial.

I'm not sure exactly what metadata you pull from git, but I can tell you how to get it from Mercurial if you list it.

Or if you open-source the metadata-getting part of the code I can write it.

**Comment by topfunky** [comment]
My initial roadblock is that Mercurial doesn't seem to offer an equivalent of Git's "git diff --numstat":

```
0       1       app/controllers/javascripts_controller.rb
11      1       app/helpers/reports_helper.rb
1       2       app/views/reports/index.haml
```

The alternative is to implement this within PeepOpen, but it could be slow to do it in Ruby, external to Mercurial.

Your suggestion of a more extensible metadata system is a good one. The PeepOpen source is available where you initially purchased and downloaded the application.

https://peepcode.com/products/peepopen

My initial task is to implement a better system for launching external commands with the user's PATH taken into account (Cocoa doesn't do this by default). Once I implement that, it would be easier to integrate other SCMs.

**Comment by sjl** [comment]
You're right, Mercurial doesn't have --numstat.  I noticed you were using that when browsing some of the other issues, and [sent a patch](http://mercurial.markmail.org/thread/5636wtcm7ripugrz) to add it, but after talking with mpm it doesn't look like it's going in.  It's trivial to do what --numstat does by counting diff lines and mpm didn't want to clutter the Mercurial codebase more.

I didn't notice that the download included the source -- that's pretty cool.  I'll take a look.

One thing though: when I click my download link it gives me 0.2.0, not 0.2.1.  Has enough changed since then that I may want to be wary of merge conflicts for you guys?

**Comment by sjl** [comment]
OK, I poked around and added support for Mercurial (and refactored just a tiny bit).  Is there somewhere I could send a patch to have it looked at?  I'm not a Ruby programmer so I'm sure it can be cleaned up a bit, but at least it works.

**Comment by sjl** [comment]
Still got the patch if you want it.

**Comment by topfunky** [comment]
Apologies for the delay. Would love to integrate the patch. Email boss@topfunky.com

**Comment by lcantey** [comment]
Any update on this?

**Comment by levifig** [comment]
Has this been integrated?

**Comment by lcantey** [comment]
It's not reporting metadata for Mercurial as of 0.3.6
